src/js/dqa-scores.jsx

```diff
@@ -0,0 +1,357 @@
+        sortOrder: 10,
+    },
+    'persistent': {
+        sortOrder: 20,
+        label: 'inherent',
+        desc: ('The issue has been investigated, ' +
+               'and it was concluded that it cannot be fixed'),
+    },
+    'withdrawn': {
+        sortOrder: 30,
+        desc: 'False alarm.  An investigation has concluded this to be a non-issue.',
+    },
+    'non-issue': {
+        sortOrder: 31,
+        desc: 'False alarm.  An investigation has concluded this to be a non-issue.',
+    },
+    'closed': {
+        sortOrder: 32,
+        desc: '',
+    },
+};
+        
+var DQAScores = React.createClass({
+    propTypes: {
+        columnWidth: React.PropTypes.number,
+        data: React.PropTypes.object,
+    },
+
+    getDefaultProps: function() {
+        return {
+            columnWidth: 120,
+            data: {},
+        };
+    },
+
+    getInitialState: function() {
+        return {
+            columnWidths: {},
+            isCollapsed: false,
+            rows: {},
+            sortBy: null,
+            sortDir: null,
+            statusList: [],
+            tableWidth: 500,
+        };
+    },
+
+    _initRows: function(props) {
+        var i;
+        var row;
+        var rows = [];
+        var score;
+        var site;
+        
+        var firstSiteData = props.data[Object.keys(props.data)[0]];
+        var statusList = Object.keys(firstSiteData);
+        var statusWeights = {}
+        
+        for (i in statusList) {
+            statusWeights[statusList[i]] = 1;
+        }
+
+        statusWeights['persistent'] = 0;
+        statusWeights['withdrawn'] = 0;
+        statusWeights['non-issue'] = 0;
+        statusWeights['closed'] = 0;
+
+        for (site in props.data) {
+            row = _.clone(props.data[site]);
+
+            score = 0;
+            for (i in row) {
+                score += row[i] * statusWeights[i];
+            }
+            row.score = score;
+            row.site = site;
+
+            rows.push(row);
+        }
+
+        // initial sort is by score, in descending order
+        var sortBy = 'score';
+        rows.sort(function(a, b) {
+            var sortVal = 0;
+
+            if (a[sortBy] > b[sortBy]) {
+                sortVal = -1;
+            }
+            else if (a[sortBy] < b[sortBy]) {
+                sortVal = 1;
+            }
+
+            return sortVal;
+        });
+    
+        return rows;
+    },
+
+    _sortRowsBy: function(sortBy) {
+        var sortDir = this.state.sortDir;
+
+        // if we are sorting by the same column as we sorted last time,
+        // reverse the sort order;
+        // otherwise pick the default order (ascending for site names,
+        // descending for scores).
+        if (sortBy === this.state.sortBy) {
+            sortDir = this.state.sortDir === SortTypes.ASC ? SortTypes.DESC : SortTypes.ASC;
+        } else {
+            if (sortBy === 'site') {
+                sortDir = SortTypes.ASC;
+            } else {
```

Comment:
This is a nit, but for some `if .. else if .. else` blocks, you separate them by a newline such as on 119. Let's keep this consistent by separating by a newline.